library/java/io/envoyproxy/envoymobile/engine/AndroidNetworkMonitor.java

```diff
@@ -104,18 +106,28 @@ public void onLost(final Network network) {
     }
   }
 
+  public static AndroidNetworkMonitor getInstance() { return instance; }
+
   @Override
   public void onReceive(Context context, Intent intent) {
     handleNetworkChange();
   }
 
+  public boolean isOnline() { return isOnline; }
```

Comment:
Should this simply be implemented as `networkType != -1`, without relying on the extra field (and babysitting it)?

Reply:
Oh shoot, yes!

Will do